plugins/modules/ecs_service_info.py

```diff
@@ -203,6 +221,26 @@ def extract_service_from(self, service):
                         e["createdAt"] = str(e["createdAt"])
         return service
 
+    def get_cluster_services(self):
+        if self.module.params["cluster"]:
+            clusters = self.module.params["cluster"]
+        else:
+            clusters = self.list_clusters()
+
+        cluster_services = {}
```

Comment:
```suggestion
        cluster_services = {}
```

Service name per cluster are unique. What will happen when more than one cluster contains the same service name?  
Will the last service name overwrite the first service name?

Reply:
I think there is an issue when the details flag is not added or set to false. This would return just the list of services with no attributes. Considering the scenario of 2 clusters, with the same HelloWorld service. 

```
- community.aws.ecs_service_info:
    details: false
 ```
 ..would get you a list of 2 service ARNs
 
```
- community.aws.ecs_service_info:
    service: arn:aws:ecs:eu-central-1:123456789012:service/my-cluster/HelloWorld
 ```
 ..would get you a single ARN 
 
 But giving service names (not service ARNs) with details: false in the current ecs_service_info returns you just the existing services. So in this scenario: 
```
- community.aws.ecs_service_info:
    details: false
    service: HelloWorld
 ```
..would in the current module fail, or force you to add a cluster and return just [HelloWorld]. But what should be the response if you have the given HelloWorld scenario? [HelloWorld] is ambiguous right? But adding cluster info or returning a cluster:service dict would be a change that changes the module response compared to the current version. 

I am not an experienced contributer. Do you have a suggestion how to proceed here?

Reply:
Ah maybe I also misinterpreted the return value

`cluster_services[cluster] = ...`

So it's already grouped by cluster. 

> returning a cluster:service dict would be a change that changes the module response compared to the current version.

I think this is the only solution. That would also mean that it is a breaking change and the changes will release with 8.0.0 and not in 7.2.0

Reply:
Okay. I need some guidance in what would be good form for the output. 

So (with details=false) the current output:
```
services:
	 - hello_world-1
	 - hello_world-2
```

If I were to slide in the clusters I should add them as an extra level below services:
```
services:
  my-cluster: 
  	 - hello_world-1
	 - hello_world-2
  my-other-cluster: 
  	 - hello_world-1
	 - hello_world-2
```

Or should this be more explicit (and renaming the root item) in the lines of: 
```
result: 
  - cluster: my-cluster
    services: 
  	 - hello_world-1
	 - hello_world-2
  - cluster: my-other-cluster
    services: 
  	 - hello_world-1
	 - hello_world-2
```

Reply:
any opinions @tremble @alinabuzachis?

Reply:
Ok, current return value for `details: false` is 

```
services:
	 - hello_world-1
	 - hello_world-2
```

using `details: true` results in 

```
services:
	 - clusterArn: ...blue
	   serviceName: hello_world-1
           ....
	 - clusterArn: ...red
	   serviceName: hello_world-1
           ...
```

I guess we should keep this structure.  
And once `cluster` name is `omit`, `details` should be automatically set to `true` 

```yml
    - name: test
      register: out
      ecs_service_info:
        cluster: "{{ omit }}"
        details: true
```

This won't break any backwards compatiblity and won't return any ambiguous results in case a service name exist in more than one cluster.

What do you think @b0tting